fe1-web/store/reducers/SocialReducer.ts

```diff
@@ -99,11 +99,29 @@ const socialSlice = createSlice({
         }
       },
     },
+
+    // Delete a chirp in the list of chirps
+    deleteChirp: {
+      prepare(laoId: Hash | string, chirp: ChirpState): any {
+        return { payload: { laoId: laoId.valueOf(), chirp: chirp } };
+      },
+      reducer(state, action: PayloadAction<{
+        laoId: string,
+        chirp: ChirpState,
+      }>) {
+        const { laoId, chirp } = action.payload;
+        const store = state.byLaoId[laoId];
+
+        if (store.byId[chirp.id]) {
+          store.byId[chirp.id] = chirp;
+        }
```

Comment:
This seems to mean that if I receive the messages out-of-order (i.e., delete first, add later), only the added version will be kept.
You might want to test (and fix) this scenario 😉

Reply:
This still looks like it needs fixing.